nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DetectDuplicateRecord.java

```diff
@@ -0,0 +1,620 @@
+//    static final AllowableValue RECORD_PATH_VALUE = new AllowableValue("entire-record", "Entire Record",
+//            "The user-defined RecordPath properties are used in the specified order to determine whether two records are equal. " +
+//                    "If the value of a RecordPath is modified or the ordering of the user-defined RecordPath properties change " +
+//                    "after processing one or more FlowFiles, duplicate detection is effectively reset.");
+
+    static final AllowableValue HASH_SET_VALUE = new AllowableValue("hash-set", "HashSet",
+            "Exactly matches records seen before with 100% accuracy at the expense of more memory usage. " +
+                    "This is not ideal for large files since a hash of each record is held in memory.");
+    static final AllowableValue BLOOM_FILTER_VALUE = new AllowableValue("bloom-filter", "BloomFilter",
+            "Space-efficient data structure ideal for large data sets uses probability to determine if a record was seen before. " +
+                    "False positive matches are possible, but false negatives are not – in other words, a query returns either \"possibly in set\" or \"definitely not in set\". " +
+                    "You should use this option if the FlowFile content is large and you can tolerate some duplication in the data.");
+
+
+    // PROPERTIES
+
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("Specifies the Controller Service to use for reading incoming data")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("record-writer")
+            .displayName("Record Writer")
+            .description("Specifies the Controller Service to use for writing out the records")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor INCLUDE_ZERO_RECORD_FLOWFILES = new PropertyDescriptor.Builder()
+            .name("include-zero-record-flowfiles")
+            .displayName("Include Zero Record FlowFiles")
+            .description("When converting an incoming FlowFile, if the conversion results in no data, "
+                    + "this property specifies whether or not a FlowFile will be sent to the corresponding relationship")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor CACHE_IDENTIFIER = new PropertyDescriptor.Builder()
+            .name("Cache The Entry Identifier")
```

Comment:
notice in the above pattern for the readers/etc  that properties often have 'name' and 'displayName' set.  The idea here is that displayName may be tweaked over time but name will stay.

Reply:
Sounds good. It should be noted this was copied from the `DetectDuplicate` processor since there is overlap in the properties offered so this issue is also present in that processor.